build/root/Makefile.generated_files

```diff
@@ -736,22 +425,16 @@ gen_openapi: $(OPENAPI_OUTFILE) $(OPENAPI_GEN)
 # For each dir in OPENAPI_DIRS, this establishes a dependency between the
 # output file and the input files that should trigger a rebuild.
 #
-# Note that this is a deps-only statement, not a full rule (see below).  This
-# has to be done in a distinct step because wildcards don't work in static
-# pattern rules.
+# Note that this is a deps-only statement, not a full rule (see below for that).
 #
 # The '$(eval)' is needed because this has a different RHS for each LHS, and
 # would otherwise produce results that make can't parse.
-#
-# We depend on the $(GOFILES_META).stamp to detect when the set of input files
-# has changed.  This allows us to detect deleted input files.
-$(foreach dir, $(OPENAPI_DIRS), $(eval                                     \
-    $(OPENAPI_OUTFILE): $(META_DIR)/$(dir)/$(GOFILES_META).stamp           \
-                                 $(gofiles__$(dir))                        \
+$(foreach dir, $(OPENAPI_DIRS), $(eval             \
+    $(OPENAPI_OUTFILE): $($(PRJ_SRC_PATH)/$(dir))  \
 ))
 
 # How to regenerate open-api code.  This emits a single file for all results.
-$(OPENAPI_OUTFILE): $(OPENAPI_GEN) $(OPENAPI_GEN)
+$(OPENAPI_OUTFILE): $(OPENAPI_GEN)
 	function run_gen_openapi() {                                                        \
```

Comment:
is there a reason to keep this shell function here (which gets immediately called)? it seems like you unrolled similar functions in the other cases.

Reply:
nope, fixed.